![imagen](Airbnb.png)

# Exercise 1

In order to begin the process of characterizing the various accommodations available on Airbnb in the city of Amsterdam, the following hypotheses are of interest:  

1. 53% of the review scores for listings on the Airbnb platform fall between 20 and 96 points. 

2. The average rental price for a property that is above the 78th percentile of city prices is 300 euros. 

3. 20% of properties in the city are being rented for less than the Amsterdam city average. 

4. Alberto claims to have rented a property on the Airbnb platform for 130 euros, which is one standard deviation above the average. 

# 1. Import libraries

In [61]:
import pandas as pd
import numpy as np

# 2. Load data

In [62]:
data= pd.read_csv('listings_m-2-.csv', sep=',')
data.head()

,id,listing_url,scrape_id,last_scraped,name,description,neighborhood_overview,picture_url,host_id,host_url,...,review_scores_communication,review_scores_location,review_scores_value,license,instant_bookable,calculated_host_listings_count,calculated_host_listings_count_entire_homes,calculated_host_listings_count_private_rooms,calculated_host_listings_count_shared_rooms,reviews_per_month
0,2818,https://www.airbnb.com/rooms/2818,2.021040e+13,12/04/2021,Quiet Garden View Room & Super Fast WiFi,Quiet Garden View Room & Super Fast WiFi<br />...,"Indische Buurt (""Indies Neighborhood"") is a ne...",https://a0.muscache.com/pictures/10272854/8dcc...,3159,https://www.airbnb.com/users/show/3159,...,10.0,9.0,10.0,NaN,t,1,0,1,0,1.90
1,20168,https://www.airbnb.com/rooms/20168,2.021040e+13,12/04/2021,Studio with private bathroom in the centre 1,17th century Dutch townhouse in the heart of t...,Located just in between famous central canals....,https://a0.muscache.com/pictures/69979628/fd6a...,59484,https://www.airbnb.com/users/show/59484,...,10.0,10.0,9.0,0363 CBB3 2C10 0C2A 1E29,t,2,0,2,0,2.50
2,25428,https://www.airbnb.com/rooms/25428,2.021040e+13,11/04/2021,"Lovely, 1 bed apt in Ctr (w.lift) -3/20-6/20(f...",Lovely apt in Centre ( lift & fireplace) near ...,NaN,https://a0.muscache.com/pictures/138431/7079a9...,56142,https://www.airbnb.com/users/show/56142,...,10.0,10.0,10.0,NaN,f,2,2,0,0,0.13
3,27886,https://www.airbnb.com/rooms/27886,2.021040e+13,11/04/2021,"Romantic, stylish B&B houseboat in canal district",Stylish and romantic houseboat on fantastic hi...,"Central, quiet, safe, clean and beautiful.",https://a0.muscache.com/pictures/02c2da9d-660e...,97647,https://www.airbnb.com/users/show/97647,...,10.0,10.0,10.0,0363 974D 4986 7411 88D8,t,1,0,1,0,1.94
4,28871,https://www.airbnb.com/rooms/28871,2.021040e+13,13/04/2021,Comfortable double room,<b>The space</b><br />In a monumental house ri...,"Flower market , Leidseplein , Rembrantsplein",https://a0.muscache.com/pictures/160889/362340...,124245,https://www.airbnb.com/users/show/124245,...,10.0,10.0,10.0,0363 607B EA74 0BD8 2F6F,f,2,0,2,0,2.59


In [63]:
data = data[['price', 'review_scores_rating']]
data.head()

,price,review_scores_rating
0,59,98.0
1,200,89.0
2,125,100.0
3,141,99.0
4,75,97.0


In [64]:
data.dtypes

price                     int64
review_scores_rating    float64
dtype: object

# 3. Hypothesis testing

#### 1. 53% of the review scores for listings on the Airbnb platform fall between 20 and 96 points.

In [68]:
#Use loc to find how many dates are between 20 - 96
between_20_96 = data.loc[(data['review_scores_rating'] >= 20) & (data['review_scores_rating'] <= 96)]
percentage = len(between_20_96) / len(data) * 100
print(f"Percentage between 20 and 96: {porcentaje:.2f}%")

Percentage between 20 and 96: 66.13%


#### The percentage between 20 and 96 is 38.39%, not 53%

#### 2. The average rental price for a property that is above the 78th percentile of city prices is 300 euros.

In [72]:
price_78 = data['price'].quantile(0.78)
average_above_78 = data.loc[data['price'] > price_78, 'price'].mean()
print(f"Price at 78th percentile: {price_78:.2f}")
print(f"Average price above 78th percentile: {average_above_78:.2f}")

Price at 78th percentile: 195.00
Average price above 78th percentile: 300.76


#### The hypothesis is correct: the average price for over 78% of the data points is 300.

#### 3. 20% of properties in the city are being rented for less than the Amsterdam city average.

In [84]:
mean_price = data['price'].mean()
below_mean = data.loc[data['price'] < mean_price]
percentage = len(below_mean) / len(data) * 100

print(f"The mean is: {mean_price:.2f}")
print(f"Count below mean: {len(below_mean)}")
print(f"The percentage of rental income below the average is: {percentage:.2f}%")

The mean is: 154.50
Count below mean: 11788
The percentage of rental income below the average is: 66.13%


#### The statement is NOT TRUE. The actual percentage is 66.13%, which means the distribution is skewed to the right.

#### 4. Alberto claims to have rented a property on the Airbnb platform for 130 euros, which is one standard deviation above the average.

In [87]:
std_price = data['price'].std()

one_std_above = mean_price + std_price

print(f"Mean + 1 std: {one_std_above:.2f}")

Mean + 1 std: 308.66


#### The statement is false for two reasons: first, the mean is 154, so 130 cannot be one standard deviation above it; and second, one standard deviation above the mean is equal to 308.